File: PersonalFinanceManager/Data Models/Transaction.cs
using System;

namespace PersonalFinanceManager.Models
{
    public class Transaction
    {
        public int Id { get; set; }
        public DateTime Date { get; set; }
        public decimal Amount { get; set; }
        public string Category { get; set; }
        public string Description { get; set; }
        public TransactionType Type { get; set; }
        
        // 添加构造函数，初始化属性
        public Transaction()
        {
            Category = string.Empty;
            Description = string.Empty;
            Date = DateTime.Now;
        }
    }
}

File: src/main/java/com/finance/demo/entity/Transaction.java
package demo.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "transactions")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Transaction {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id; // 改为 Long

    @Column(nullable = false, name = "transaction_date")
    private LocalDateTime date;

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal amount;

    @Column(nullable = false, length = 100)
    private String category;

    @Column(length = 500)
    private String description;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 10)
    private TransactionType type;

    // 添加用户关联
    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;
}